src/btree/bt_curprev.c

```diff
@@ -766,6 +766,9 @@ __wt_btcur_prev(WT_CURSOR_BTREE *cbt, bool truncating)
         if (F_ISSET(cbt, WT_CBT_READ_ONCE))
             LF_SET(WT_READ_WONT_NEED);
 
+        if (!F_ISSET(session->txn, WT_TXN_HAS_SNAPSHOT))
+            LF_SET(WT_READ_VISIBLE_ALL);
+
```

Comment:
Should this assert we have a snapshot? That is, are we planning to fully support code paths where we don't have a snapshot, or are we planning to turn them on only by demand?

I'm slightly in favor of the latter, that is, asserting we have a snapshot until we know there's a code path where we legitimately don't have a snapshot. I could be talked out of that, though.

Edit -- from later changes, it looks to me like this is a path where you did identify a caller that doesn't have a snapshot transaction? In which case, please ignore my comment.

Reply:
rollback to stable and stats cursor are the ones I found.